### PR TITLE
Extended catalog

### PR DESCRIPTION
**Pull request by jjkester**

### CODE REVIEW

**Comment by jjkester** [comment]
We probably want this in the demo as well?

File: Code/src/main/java/nl/utwente/group10/ui/Main.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public void start(Stage stage) throws Exception {
         stage.setOnCloseRequest(event -> System.exit(0));
         stage.setScene(scene);
 
-        stage.setMaximized(true);
+        //stage.setMaximized(true);
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/group10/ui/Main.java]: This file's changes seem to be just Mac-specific hotfixes. Did the Mac crew come up with a solution yet? (@Kristel-92 @TheMSB)

File: Code/src/main/resources/catalog/catalog.xml [hunk 2]
```diff
@@ -18,105 +35,584 @@
...
+        	</function>
+        	<function name="uncurry" signature="(a -> b -> c) -> (a, b) -> c">
+        		uncurry converts a curried function to a function on pairs
+        	</function>
+        </category>
+        <category name="Equality">
+        	<function name="(==)" signature="Eq a => a -> a -> Bool"></function>
+        	<function name="(/=)" signature="Eq a => a -> a -> Bool"></function>
+        </category>
+        <category name="Ordered Equality">
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Combine this with Equality?

**Comment by jjkester** [comment]
I just assumed this branch only contained catalog stuff. Silly me :cry:

File: Code/src/main/resources/catalog/catalog.xml (hunk 2 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: Kritel told me that she used the same categories as Hoogle, so we might stick with that. Although equality is very empty right now.

File: Code/src/main/resources/catalog/catalog.xml [hunk 3]
```diff
@@ -18,105 +35,584 @@
...
+        	<function name="round" signature="(RealFrac a, Integral b) => a -> b">
+        		round x returns the nearest integer to x; the even integer if x is equidistant between two integers.
+        	</function>
+        	<function name="ceiling" signature="(RealFrac a, Integral b) => a -> b">
+        		ceiling x returns the least integer not less than x.
+        	</function>
+        	<function name="floor" signature="(RealFrac a, Integral b) => a -> b">
+        		floor x returns the greatest integer not greater than x.
+        	</function>
+        	<function name="floatRadix" signature="RealFloat a => a -> Integer">
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Are the functions from here up to and including scaleFloat interesting enough to include here?

File: Code/src/main/resources/catalog/catalog.xml [hunk 4]
```diff
@@ -18,105 +35,584 @@
...
+        	<function name="atan2" signature="RealFloat a => a -> a -> a">
+        		a version of arctangent taking two real floating-point arguments. For real floating x and 
+        		y, atan2 y x computes the angle (from the positive x-axis) of the vector from the 
+        		origin to the point (x,y). atan2 y x returns a value in the range [-pi, pi]. It follows 
+        		the Common Lisp semantics for the origin when signed zeroes are supported. atan2 y 
+        		1, with y in a type that is RealFloat, should return the same value as atan y. A 
+        		default definition of atan2 is provided, but implementors can provide a more accurate 
+        		implementation.
+        	</function>
+        	<function name="subtract" signature="Num a => a -> a -> a">
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Do we need this one? Doesn't it do the same as (-) in our setup?

File: Code/src/main/resources/catalog/catalog.xml [hunk 5]
```diff
@@ -18,105 +35,584 @@
...
+        	<function name="isInfinite" signature="RealFloat a => a -> Bool">
+        		True if the argument is an IEEE infinity or negative infinity
+        	</function>
+        	<function name="isDenormalized" signature="RealFloat a => a -> Bool">
+        		True if the argument is too small to be represented in normalized format.
+        	</function>
+        	<function name="isNegativeZero" signature="RealFloat a => a -> Bool">
+        		True if the argument is an IEEE negative zero.
+        	</function>
+        	<function name="isIEEE" signature="RealFloat a => a -> Bool">
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Do we need this one? Doesn't it always return True?

File: Code/src/main/resources/catalog/catalog.xml [hunk 6]
```diff
@@ -18,105 +35,584 @@
...
+        		True if the argument is an IEEE floating point number.
+        	</function>
+        	<function name="atan2" signature="RealFloat a => a -> a -> a">
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Whats the difference between this and atan?

File: Code/src/main/resources/catalog/catalog.xml [hunk 7]
```diff
@@ -18,105 +35,584 @@
...
+        		constant function.
+        	</function>
+        	<function name="(.)" signature="(b -> c) -> (a -> b) -> a -> c">
+        		Function composition
+        	</function>
+        	<function name="flip" signature="(a -> b -> c) -> b -> a -> c">
+        		flip f takes its (first) two arguments in the reverse order of f.
+        	</function>
+        	<function name="($)" signature="(a -> b) -> a -> b">
+        		Application operator. This operator is redundant, since ordinary application (f x) means
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Description doesn't make much sense in viskell: the operator is not redundant, it's the primary way to do function application.

File: Code/src/main/resources/catalog/catalog.xml (hunk 4 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: Guess so. I think we can remove this.

File: Code/src/main/resources/catalog/catalog.xml [hunk 8]
```diff
@@ -18,105 +35,584 @@
...
+        	</function>
+        	<function name="until" signature="(a -> Bool) -> (a -> a) -> a -> a">
+        		until p f yields the result of applying f until p holds.
+        	</function>
+        	<function name="asTypeOf" signature="a -> a -> a">
+        		asTypeOf is a type-restricted version of const. It is usually used an infix operator, and
+        		its typing forces its first argument (which is usually overloaded) to have the same type as the
+        		second.
+        	</function>
+        	<function name="error" signature="[Char] -> a">
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Displays an error message where? We don't show runtime errors anywhere.

File: Code/src/main/resources/catalog/catalog.xml (hunk 5 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: We might. Again, Kristel tried to use all of Prelude, so why not leave it in.

File: Code/src/main/resources/catalog/catalog.xml [hunk 9]
```diff
@@ -18,105 +35,584 @@
...
+        	</function> 
+        	<function name="undefined" signature="a">
+        		A special case of error. It is expected that compilers will recognize this and insert error 
+        		messages which are more appropriate to the context in which undefined appears.
+        	</function>
+        	<function name="seq" signature="a -> b -> b">
+        		The value of seq a b is bottom if a is bottom, and otherwise equal to b. seq is usually 
+        		introduced to improve performance by avoiding unneeded laziness.
+        	</function>
+        	<function name="($!)" signature="(a -> b) -> a -> b">
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: What does this do?

File: Code/src/main/resources/catalog/catalog.xml [hunk 10]
```diff
@@ -18,105 +35,584 @@
...
+        	<function name="unlines" signature="[String] -> String">
+	        	unlines is an inverse operation to lines. It joins lines, after appending a
+	        	terminating newline to each.
+        	</function>
+        	<function name="unwords" signature="[String] -> String">
+        		unwords is an inverse operation to words. It joins words with separating spaces.
+        	</function>
+        </category>
+        <category name="Converting to and from String">
+        	<function name="showsPrec" signature="Show a => Int -> a -> ShowS">
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: ShowS typeclass is not listed as a typeclass.

File: Code/src/main/resources/catalog/catalog.xml [hunk 11]
```diff
@@ -18,105 +35,584 @@
...
+        	<function name="showList" signature="Show a => [a] -> ShowS">
+        		The method showList is provided to allow the programmer to give a
+        		specialised way of showing lists of values. For example, this is used by the
+        		predefined Show instance of the Char type, where values of type String
+        		should be shown in double quotes, rather than between square brackets.
+        	</function>
+        	<function name="shows" signature="Show a => a -> ShowS">
+        		equivalent to showsPrec with a precedence of 0.
+        	</function>
+        	<function name="showChar" signature="Char -> ShowS">
```
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Do we need the following three utility functions?

File: Code/src/main/resources/catalog/catalog.xml (hunk 6 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: https://hackage.haskell.org/package/base-4.7.0.0/docs/Prelude.html#v:atan2

File: Code/src/main/resources/ui/DisplayBlock.fxml [hunk 12]
```diff
@@ -4,18 +4,10 @@
...
-        </top>
-        <center>
-        <VBox styleClass="display, block" spacing="12">
-            <Label id="label_header" text="OUTPUT" styleClass="header" alignment="center_left" />
-            <Label text="${controller.output}" styleClass="title" alignment="center" />
+        <VBox styleClass="display, block" spacing="12" alignment = "center">
+        	<FlowPane fx:id="anchorSpace" prefWrapLength="100" vgap="50" hgap="10" alignment="center"/>
+            <Label id="label_header" text="OUTPUT" styleClass="header" />
+            <Label text="${controller.output}" styleClass="title" />
+            <!-- <FlowPane fx:id="outputSpace" alignment = "center"/> -->
```
Comment (wandernauta) [inline Code/src/main/resources/ui/DisplayBlock.fxml]: Please no commented-out code: either remove it or don't.

File: Code/src/main/resources/ui/FunctionBlock.fxml [hunk 13]
```diff
@@ -25,4 +43,5 @@
             <VBox fx:id="outputSpace" alignment = "center"/>
         </bottom>
     </BorderPane>
+     -->
```
Comment (wandernauta) [inline Code/src/main/resources/ui/FunctionBlock.fxml]: Same here: delete or keep, don't comment.

File: Code/src/main/resources/catalog/catalog.xml (hunk 7 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: Can fix that. Probably copied from Hoogle.

File: Code/src/main/resources/catalog/catalog.xml (hunk 8 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: Better to remove I guess?

File: Code/src/main/resources/catalog/catalog.xml (hunk 9 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: https://hackage.haskell.org/package/base-4.7.0.0/docs/Prelude.html#v:seq

File: Code/src/main/resources/catalog/catalog.xml (hunk 10 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: Error by Kristel, has to be added.

File: Code/src/main/resources/ui/DisplayBlock.fxml (hunk 12 above)
Comment (jjkester) [inline Code/src/main/resources/ui/DisplayBlock.fxml]: Do not know what this is.

File: Code/src/main/resources/ui/FunctionBlock.fxml (hunk 13 above)
Comment (jjkester) [inline Code/src/main/resources/ui/FunctionBlock.fxml]: Again, do not know what this is.

File: Code/src/main/java/nl/utwente/group10/ui/Main.java (hunk 1 above)
Comment (jjkester) [inline Code/src/main/java/nl/utwente/group10/ui/Main.java]: I am going to remove these changes for this PR.

File: Code/src/main/resources/catalog/catalog.xml (hunk 2 above)
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Yes, my reasoning was that people are unlikely to have the Hoogle categories memorized, and the small fraction who do will probably not have much problems finding the Equality category when they're looking for the Ordered Equality category.

File: Code/src/main/resources/catalog/catalog.xml (hunk 8 above)
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: I'd say so.

File: Code/src/main/resources/catalog/catalog.xml (hunk 3 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: I think you're right. Removed.

File: Code/src/main/resources/catalog/catalog.xml (hunk 10 above)
Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: Except that ShowS is a type here, not a typeclass.

**Comment by jjkester** [comment]
Fixed a lot of stuff. I think we can merge this (to master, and probably demo as well).

File: Code/src/main/resources/catalog/catalog.xml (hunk 10 above)
Comment (wandernauta) [inline Code/src/main/resources/catalog/catalog.xml]: Apparently, ShowS is really `String -> String` - you could put that here directly, or drop the ShowS functions, your call.

Comment (jjkester) [inline Code/src/main/resources/catalog/catalog.xml]: Do we do anything with the output?

**Comment by wandernauta** [comment]
Merged to `master`, thanks!

**Pull request merged**